cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/ConfigurationFinder.java

```diff
@@ -29,6 +33,44 @@ public class ConfigurationFinder {
   /** Logger instance */
   private static final Logger LOG = LoggerFactory.getLogger(ConfigurationFinder.class);
 
+  /**
+   * load properties from .properties file into TemplateSetConfiguration if found valid properties otherwise load
+   * default values
+   *
+   * @param path to a .properties file
+   * @return TemplateSetConfiguration instance
+   */
+  public static TemplateSetConfiguration loadTemplateSetConfigurations(Path path) {
+
+    Properties props = new Properties();
+    try {
+      props = readConfigurationFile(path);
+    } catch (CobiGenRuntimeException e) {
+
+    }
+
+    String groupId = ConfigurationConstants.CONFIG_PROPERTY_TEMPLATE_SETS_GROUPIDS;
+    String snapshot = ConfigurationConstants.CONFIG_PROPERTY_TEMPLATE_SETS_SNAPSHOTS;
+    String hide = ConfigurationConstants.CONFIG_PROPERTY_TEMPLATE_SETS_HIDE;
+    String disableLookup = ConfigurationConstants.CONFIG_PROPERTY_TEMPLATE_SETS_DISABLE_LOOKUP;
+    String defaultGroupId = ConfigurationConstants.CONFIG_PROPERTY_TEMPLATE_SETS_DEFAULT_GROUPID;
+
+    List<String> groupIdsList = (props.getProperty(groupId) != null)
+        ? Arrays.asList(props.getProperty(groupId).split(","))
+        : new ArrayList<>();
+    // Creating a new ArrayList object which can be modified
+    List<String> groupIds = new ArrayList<>(groupIdsList);
+    if (props.getProperty(disableLookup) == null || props.getProperty(disableLookup).equals("false"))
+      groupIds.add(defaultGroupId);
+
+    boolean useSnapshots = props.getProperty(snapshot) == null || props.getProperty(snapshot).equals("false") ? false
+        : true;
```

Comment:
as this expressions results in true or false the  expression could be used directly like this:
boolean useSnapshots = !(props.getProperty(snapshot) == null || props.getProperty(snapshot).equals("false"));

Reply:
When trying to debug invalidConfigurationTest with this expression, for some unknown reason an InvokationTargetexception will be thrown.